spec/system/admin/products_v3/actions_spec.rb

```diff
@@ -475,6 +475,39 @@ def expect_other_columns_visible
         end
       end
     end
+
+    describe "Preview" do
+      let(:product) { create(:product, name: "Apples") }
+      let!(:variant) { create(:variant, product:) }
+
+      it "show product preview modal" do
+        login_as_admin
+        visit admin_products_url
+
+        within row_containing_name("Apples") do
+          open_action_menu
+          click_link "Preview"
+        end
+
+        expect(page).to have_content("Product preview")
+
+        within "#product-preview-modal" do
+          # Shop tab
+          expect(page).to have_selector("h3 a span", text: "Apples")
+          add_buttons = page.all(".add-variant")
+          expect(add_buttons.length).to eql(2)
+
+          # Product Details tab
+          find("a", text: "Product details").click # click_link doesn't work
+          expect(page).to have_selector("h3", text: "Apples")
```

Comment:
❓ This might match the same condition as the other tab, so may not guarantee the tab changed. Is there another detail we can check for?

Reply:
It's a slightly different check than the other one, the shop tab is looking for "Apples" in a `span` in a `a` in a `h3` :
```
expect(page).to have_selector("h3 a span", text: "Apples")
```

Reply:
Yes, but I think that the `text` option looks at the combined text of all child nodes, hence `have_selector("h3", text: "Apples")` would match `<h3><a><span>Apples` (I think).

So maybe we can check for the product description also?